simulator/syncer/resource.go

```diff
@@ -0,0 +1,122 @@
+}
+
+// MandatoryFilteringFunctions is a map of GroupVersionResource to FilteringFunction.
+// Outside users must call them in their own filtering functions.
+var MandatoryFilteringFunctions = map[schema.GroupVersionResource]FilteringFunction{
+	{Group: "", Version: "v1", Resource: "pods"}: filterPods,
+}
+
+// FilteringFunction is a function that filters a resource.
+// If it returns false, the resource will not be imported.
+type FilteringFunction func(ctx context.Context, resource *unstructured.Unstructured, clients *Clients, event Event) (bool, error)
+
+// MutatingFunction is a function that mutates a resource before importing it.
+type MutatingFunction func(ctx context.Context, resource *unstructured.Unstructured, clients *Clients, event Event) (*unstructured.Unstructured, error)
+
+func mutatePV(ctx context.Context, resource *unstructured.Unstructured, clients *Clients, _ Event) (*unstructured.Unstructured, error) {
+	var pv v1.PersistentVolume
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(resource.UnstructuredContent(), &pv)
+	if err != nil {
+		return nil, err
+	}
+
+	if pv.Status.Phase == v1.VolumeBound {
+		// PersistentVolumeClaims's UID is changed in a destination cluster when importing from a source cluster,
+		// and thus we need to update the PVC UID in the PersistentVolume.
+		// Get PVC of pv.Spec.ClaimRef.Name.
+		pvc, err := clients.SrcDynamicClient.Resource(schema.GroupVersionResource{
+			Group:    "",
+			Version:  "v1",
+			Resource: "persistentvolumeclaims",
+		}).Namespace(pv.Spec.ClaimRef.Namespace).Get(ctx, pv.Spec.ClaimRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		pv.Spec.ClaimRef.UID = pvc.GetUID()
+	}
+
+	modifiedUnstructed, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&pv)
+	return &unstructured.Unstructured{Object: modifiedUnstructed}, err
+}
+
+func mutatePods(_ context.Context, resource *unstructured.Unstructured, _ *Clients, _ Event) (*unstructured.Unstructured, error) {
+	var pod v1.Pod
+	err := runtime.DefaultUnstructuredConverter.FromUnstructured(resource.UnstructuredContent(), &pod)
+	if err != nil {
+		return nil, err
+	}
+
+	// Pods must have the default ServiceAccount because ServiceAccount is not synced.
+	// But static pods must not have a ServiceAccount.
+	if pod.Spec.ServiceAccountName != "" {
+		pod.Spec.ServiceAccountName = "default"
+	}
+
+	// If the pod has an owner, it may be deleted because resources such as ReplicaSet are not synced.
+	pod.OwnerReferences = nil
+
+	modifiedUnstructed, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&pod)
+	return &unstructured.Unstructured{Object: modifiedUnstructed}, err
+}
+
+// filterPods checks if a pod is already scheduled when it's updated.
+// We only want to update pods that are not yet scheduled.
+func filterPods(_ context.Context, resource *unstructured.Unstructured, _ *Clients, event Event) (bool, error) {
+	if event != Update {
```

Comment:
It's more intuitive.

```suggestion
	if event == Add {
		// We always add a Pod, regardless it's scheduled or not.
```